registry/auth_test.go

```diff
@@ -24,6 +24,14 @@ func TestEncodeAuth(t *testing.T) {
 	if authStr != "a2VuOnRlc3Q=" {
 		t.Fatal("AuthString encoding isn't correct.")
 	}
+	username0, password0, _ := decodeAuth("dXNlcm5hbWU6cGFzc3dvcmQ=")
+	if username0 != "username" || password0 != "password" {
```

Comment:
Oh I was just saying if err!= nil t.Fatal(err). These  cases you newly added because of my comment seems unnecessary. :-)